sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ResourceDataWriteProxy.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.ClientModel.Tests.Client.Models.ResourceManager.Resources;
+using System.Collections.Generic;
+using System.Text.Json;
+using ClientModel.Tests.ClientShared;
+
+namespace System.ClientModel.Tests.Client
+{
+    public class ResourceDataWriteProxy : IJsonModel<ResourceProviderData>
```

Comment:
Curious -- it's called WriteProxy ... is it only for writing or could it be used for reading as well?